src/main/java/org/jvnet/hudson/test/InboundAgentRule.java

```diff
@@ -50,32 +52,184 @@
  */
 public final class InboundAgentRule extends ExternalResource {
 
-    private final Map<String, Process> procs = new HashMap<>();
+    private final ConcurrentMap<String, Process> procs = new ConcurrentHashMap<>();
+
+    /**
+     * The options used to (re)start an inbound agent.
+     */
+    public static class Options {
+        // TODO Java 14+ use records
+
+        @CheckForNull private String name;
```

Comment:
personal preference would be to move the annotation above it, it looks strange to me as it's in the same grouping of variables and feels mis-aligned. (feel free to ignore)

```suggestion
        @CheckForNull
        private String name;
```

Reply:
`google-java-format` puts them on the same line if the whole line would be within the margin, and I have seen another argument that they should be on the same line because type annotations should be close to their corresponding type. I do not care much either way.